plugin/web/sharedComponents/ModifiedImage/hooks.ts

```diff
@@ -0,0 +1,77 @@
+import { useState, useEffect } from "react";
+import tinycolor from "tinycolor2";
+
+type Props = {
+  imageUrl: string;
+  blendColor: string;
+  width: number;
+  height: number;
+};
+
+const useModifiedImage = ({ imageUrl, blendColor, width, height }: Props) => {
+  const [modifiedImageUrl, setModifiedImageUrl] = useState("");
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const image = new Image();
+    image.crossOrigin = "anonymous";
+    image.src = imageUrl;
+    image.onload = () => {
+      if (!image.complete) {
+        console.error("Failed to load image due to cross-origin error");
+        setLoading(false);
+        return;
+      }
+      const canvas = document.createElement("canvas");
+      canvas.width = width;
+      canvas.height = height;
+      const ctx = canvas.getContext("2d");
+      if (!ctx) {
+        setLoading(false);
+        return;
+      }
+      ctx.drawImage(image, 0, 0, width, height);
+
+      const mappedColor = tinycolor(blendColor).toRgb();
+      const color = `rgba(${mappedColor.r}, ${mappedColor.g}, ${mappedColor.b}, ${mappedColor.a})`;
+
+      // Convert the canvas to a data URL
+      const canvas2 = document.createElement("canvas");
+      canvas2.width = width;
+      canvas2.height = height;
+      const ctx2 = canvas2.getContext("2d");
+      if (!ctx2) {
+        setLoading(false);
+        return;
+      }
+      ctx2.drawImage(image, 0, 0, width, height);
+      ctx2.globalCompositeOperation = "multiply";
+      ctx2.fillStyle = color;
+      ctx2.fillRect(0, 0, canvas2.width, canvas2.height);
+      ctx2.globalCompositeOperation = "destination-in";
+      ctx2.drawImage(image, 0, 0, width, height);
+
+      const modifiedImage = new Image();
+      modifiedImage.src = canvas2.toDataURL();
+
+      setModifiedImageUrl(modifiedImage.src);
```

Comment:
Is there any chance that modifiedImage.src will be undefined or any other issues? I feel like displaying the original image is good to fallback to

Reply:
There is a chance that modifiedImage.src could be undefined in case of some issue while creating the modified image. For example, if the canvas element is not able to draw the image due to some error, the modifiedImage.src could be undefined.

Reply:
Done